src/class-plugin-curated-posts.php

```diff
@@ -74,9 +74,12 @@ public function with_query_context( array $context, array $attributes, WP_Block_
 		}
 
 		$pinned_posts = $attributes['posts'] ?? $block_type->attributes['posts']['default'];
-		$pinned_posts = array_filter( $pinned_posts, function ( $id ) {
-			return 'publish' === get_post_status( $id );
-		} );
+		$pinned_posts = array_map( function ( $id ) {
+			if ( ! $id ) {
+				return null;
+			}
+			return 'publish' === get_post_status( $id ) ? $id : null;
+		}, $pinned_posts );
```

Comment:
logic can be simplified to `$id && 'publish' === get_post_status( $id )`... which means this could use an arrow function if you felt so inclined:

```suggestion
		$pinned_posts = array_map( fn ( $id ) => $id && 'publish' === get_post_status( $id ) ? $id : null, $pinned_posts );
```